### PR TITLE
Removes debug settings from wasm32_unknown_emscripten default link args

### PR DESCRIPTION
**Pull request by hoodmane**
This is a debug setting. We should only make debug builds if user requests
a debug build. Currently this is inserted in release builds.

Furthermore, it would be better to insert these settings in --pre-link-args
because then it would be possible to override them if appropriate. Because
these are inserted at the end, it is necessary to patch emscripten to remove
them.

@sbc100

### CODE REVIEW

File: compiler/rustc_target/src/spec/wasm32_unknown_emscripten.rs [hunk 1]
```diff
@@ -20,8 +20,6 @@ pub fn target() -> Target {
             "-s".into(),
             "ERROR_ON_UNDEFINED_SYMBOLS=1".into(),
```
Comment (sbc100) [inline compiler/rustc_target/src/spec/wasm32_unknown_emscripten.rs:21]: you can also remove this line .. since it is now the default.

**Review by sbc100 — COMMENTED** [review]
@tlively who worked on this toolchain

File: compiler/rustc_target/src/spec/wasm32_unknown_emscripten.rs [hunk 2]
```diff
@@ -17,10 +17,6 @@ pub fn target() -> Target {
     post_link_args.insert(
         LinkerFlavor::Em,
         vec![
-            "-s".into(),
-            "ERROR_ON_UNDEFINED_SYMBOLS=1".into(),
-            "-s".into(),
-            "ASSERTIONS=1".into(),
             "-s".into(),
             "ABORTING_MALLOC=0".into(),
```
Comment (sbc100) [inline compiler/rustc_target/src/spec/wasm32_unknown_emscripten.rs:21]: Why you are there you could also merge these two lines into a single `-sABORTING_MALLOC=0".into(),`?

**Review by tlively — APPROVED** [review]
This seems reasonable to me.

**Pull request merged**